pkg/cmdline/filters.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package cmdline
+
+import (
+	"fmt"
+	"strings"
+)
+
+// RemoveFilter filters out variable for a given space-separated kernel commandline
+func removeFilter(input string, variables []string) string {
+	var newCl []string
+
+	// kernel variables must allow '-' and '_' to be equivalent in variable
+	// names. We will replace dashes with underscores for processing as
+	// `doParse` is doing.
+	for i, v := range variables {
+		variables[i] = strings.Replace(v, "-", "_", -1)
+	}
+
+	doParse(input, func(flag, key, canonicalKey, value, trimmedValue string) {
+		skip := false
+		for _, v := range variables {
+			if canonicalKey == v {
+				skip = true
+				break
+			}
+		}
+		if skip {
+			return
+		}
+		newCl = append(newCl, flag)
+	})
+	return strings.Join(newCl, " ")
+}
+
+// Filter represents and kernel commandline filter
+type Filter interface {
```

Comment:
This interface has only one implementation -- updater. You could get rid of this interface, and just s/updater/Filter